src/middleware/validate.ts

```diff
@@ -0,0 +1,34 @@
+import { NextFunction, Request, Response } from 'express';
+import { logger, sendResponse } from '../utils';
+import Joi from 'joi';
+
+enum requestType {
+  body='body',
+  params='params',
+  queries='queries'
+}
+
+export const validationMiddleware = (schema: Joi.ObjectSchema, type: requestType = requestType.body)=>(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+
+  try {
+    const { error } = schema.validate(req.body, { abortEarly: false })
+
+    if(error){
+      const context = (error.details[0] as any).context.message;
+      const errorMessage = context? context : error.message;
+      
+      console.log(error);
```

Comment:
here you will return only one message but when we have alot of errors what will happened?? and remember to remove console.log

Reply:
Done 👍